### DISCUSSION TITLE
OLED display?

### DISCUSSION THREAD

**Discussion by ustajan** [SpenceKonde/ATTinyCore#828] [Support / Q & A]
Does ATTinyCore support operation of an OLED display (through the SDA and SCL pins)?  
If not, can people suggest alternative approaches / hacks ?

Much thanks.

**Comment by hmeijdam** [comment]
I am using the [Tiny4kOLED library](https://github.com/datacute/Tiny4kOLED) in combination with an Attiny85. Works just fine.

**Reply by ustajan** [reply]
Thanks for the pointer!  I got Tiny4kOLED working.
It's simple enough...but seems to be VERY slow.  Every screen write takes >400 milliseconds to complete.  Is that also your experience?  Is this perhaps normal for a tiny machine like attiny85?  [Here's more detail ](https://github.com/datacute/Tiny4kOLED/discussions/60#discussion-5947471)about this topic.

**Reply by hmeijdam** [reply]
I don't recall it was slow, or I didn't bother. What clock speed have you clocked your ATTiny?

**Reply by hmeijdam** [reply]
I am getting 136 when clocked at 8MHz internal in the I2CSpeedTest sketch that is part of the Tiny4K library.

![image](https://github.com/SpenceKonde/ATTinyCore/assets/9595465/2b468618-6a7f-4b31-80b2-b7fef83cb4a3)

I am getting 123 when clocked at 16MHz and 367 when clocked at 1MHz.

All with the standard "Wire.h"

With the  `#include <Tiny4kOLED_bitbang.h>` it gets quite a bit faster. Then I get 45ms to fill the screen.
I recommend you play a bit with that I2CSpeedTest sketch.

**Reply by ustajan** [reply]
Oh wow...`#include <Tiny4kOLED_bitbang.h>` really sped it up!!!   Now `oled.clear()` takes 47 ms,  and a single row write takes up mere additional 18ms!  Thank you thank you thank you, @hmeijdam !!! 

Now I have to ask -- why wasn't `_bitbang.h`  offered as the default library?  Does the "regular" `Tiny4kOLED.h` have something that `_bitbang.h` doesn't...?

**Reply by hmeijdam** [reply]
You would have to search for an answer in the Tiny4KOLED forum. Maybe compatibility reasons.
